astropy/io/votable/tests/test_vo.py

```diff
@@ -266,6 +266,8 @@ def test_iter_coosys(self):
 )
 def test_select_missing_columns_error_message(columns, expected_missing):
     # see https://github.com/astropy/astropy/pull/15956
+    # Turn into numpy strings (otherwise match fails on numpy >= 2.0).
+    expected_missing = [np.str_(e) for e in expected_missing]
```

Comment:
I'm don't think this is the correct fix here. I'll link to my own solution when I open the PR.

Reply:
https://github.com/astropy/astropy/pull/16442